mongodbatlas/resource_mongodbatlas_database_user.go

```diff
@@ -289,15 +289,15 @@ func resourceMongoDBAtlasDatabaseUserImportState(d *schema.ResourceData, meta in
 
 	d.SetId(encodeStateID(map[string]string{
 		"project_id":         *projectID,
-		"username":           *username,
-		"auth_database_name": *authDatabaseName,
+		"username":           u.Username,
+		"auth_database_name": u.DatabaseName,
 	}))
 
 	return []*schema.ResourceData{d}, nil
 }
 
 func splitDatabaseUserImportID(id string) (projectID, username, authDatabaseName *string, err error) {
-	var re = regexp.MustCompile(`(?s)^([0-9a-fA-F]{24})-(.*)-([a-z]{1,15})$`)
+	var re = regexp.MustCompile(`(?s)^([0-9a-fA-F]{24})-(.*)-([$a-z]{1,15})$`)
```

Comment:
Just an FYI the admin database can only be admin or $external so could possibly simplify this, not required but if we run into more issues may make sense.

Reply:
I made a test validation in both cases (with `admin` and `$external`) and they worked fine. I decided to leave in this way because in the future if `$external` change or add a new for example `$internal` it will work fine.

Reply:
Not a bad idea to future proof it!  Thank you for testing and agree.